### PR TITLE
feat(broker): support timer start event

### LINKED ISSUE #1787: Support timer start events
Depends on #1785

### PR DESCRIPTION
**Pull request by miguelpires**
Implements timer start events with the restriction that there can be only one per workflow.

closes #1787

### CODE REVIEW

**Review by saig0 — CHANGES_REQUESTED** [review]

**Comment by MiguelPires** [comment]
I made some changes incorporating your feedback. This commit does a couple of things:

- Refactors the tests
- Only creates the timer in the partition with the lowest id
- Uses the workflow key instead of the bpmn id to avoid rescheduling the timer with the new properties
- Creates and removes timers using existing processors/methods

**Review by saig0 — CHANGES_REQUESTED** [review]

File: broker-core/src/test/java/io/zeebe/broker/workflow/TimerStartEventTest.java [hunk 1]
```diff
@@ -244,17 +234,18 @@ public void shouldReplaceTimerStartWithNoneStart() {
             .getLast()
```
Comment (saig0) [inline broker-core/src/test/java/io/zeebe/broker/workflow/TimerStartEventTest.java]: Same here, the test waits 5 seconds because of `getLast()`.

**Review by saig0 — CHANGES_REQUESTED** [review]
Almost perfect. Just a few tiny things to speed up the tests.

File: broker-core/src/test/java/io/zeebe/broker/workflow/TimerStartEventTest.java [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+            .timerWithCycle("R2/PT4S")
+            .endEvent("end_4")
+            .done();
+    testClient.deploy(slowerModel);
+
+    // then
+    assertThat(RecordingExporter.timerRecords(TimerIntent.CANCELED).getFirst()).isNotNull();
+
+    timerRecord =
+        RecordingExporter.timerRecords(TimerIntent.CREATED).limit(3).asList().get(2).getValue();
```
Comment (saig0) [inline broker-core/src/test/java/io/zeebe/broker/workflow/TimerStartEventTest.java]: This is a bit shorter:

```suggestion
        RecordingExporter.timerRecords(TimerIntent.CREATED).skip(2).getFirst().getValue();
```

**Review by saig0 — CHANGES_REQUESTED** [review]

**Review by saig0 — APPROVED** [review]

**Comment by MiguelPires** [comment]
bors cancel

**Pull request merged**